### PR TITLE
fix decorrelate with limit

### PR DESCRIPTION
**Pull request by zinking**
comments in jura

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java [hunk 1]
```diff
@@ -777,9 +777,11 @@ private static Expression implementNullSemantics(
       // v0 != null && v1 != null && f(v0, v1)
       for (Ord<RexNode> operand : Ord.zip(call.getOperands())) {
         if (translator.isNullable(operand.e)) {
-          list.add(
-              translator.translate(
-                  operand.e, NullAs.IS_NOT_NULL));
+          Expression te = translator.translate(operand.e, NullAs.IS_NOT_NULL);
+          Class tec = (Class) te.getType();
+          if (!tec.isPrimitive() || tec.equals(boolean.class)) {
```
Comment (vlsi) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: Why do you ignore certain types?

Comment (zinking) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java]: @vlsi I moved the comments into JIRA.  the issue is manifested as `byte` object is  emitted into the    `not null` expression. but `byte` primitive object itself is already `not null`. so primitives should be skipped. 

anyways Julian mentioned this is not worth fixing using this approach at the moment.

**Comment by julianhyde** [comment]
Which JIRA case covers this? Can you please add a link?

If you don't think this code is good to commit, please close this PR.

**Comment by zinking** [comment]
https://issues.apache.org/jira/browse/CALCITE-709

**Pull request closed without merging**